modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -226,10 +232,20 @@ public TableManager(
 
             return node.id();
         };
+
         clusterNodeResolver = topologyService::getByAddress;
 
         tablesByIdVv = new VersionedValue<>(null, HashMap::new);
 
+        registry.accept(token -> {
+            List<CompletableFuture<?>> futures = new ArrayList<>(beforeTablesVvComplete);
+
+            beforeTablesVvComplete.clear();
+
+            return CompletableFuture.allOf(futures.toArray(new CompletableFuture[] {}))
+                    .thenRun(() -> tablesByIdVv.complete(token));
```

Comment:
Well, not sure, but I'd also add orTimeout(some rather big amount of time) terminator for diagnostic purposes that will fail an exception explicitly with some extra output about futures and context. I believe, that we should create corresponding ticket for removing such diagnostic timeouts and left such ticket within todo here. It's up to you though.